bundles/ch.elexis.core.ui/src/ch/elexis/core/ui/views/rechnung/RechnungsBlatt.java

```diff
@@ -890,4 +889,30 @@ public void display() {
 		}
 	}
 
+	private List<String> combineAndSortTrace(List<String> trace, List<String> mandatorTrace) {
+		SimpleDateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy, HH:mm:ss");
+		return trace.stream().map(statusEntry -> {
+			String[] stm = statusEntry.split("\\s*:\\s", 2);
+			String statusText = stm[0] + " : " + InvoiceState.fromState(Integer.parseInt(stm[1])).getLocaleText();
+			String mandatorLabel = mandatorTrace.stream().filter(m -> m.contains(stm[0])).findFirst().map(m -> {
+				mandatorTrace.remove(m);
+				if (m.contains(":")) {
+					String[] parts = m.split(":");
+					return parts.length > 1 ? parts[parts.length - 1].trim() : "";
+				}
+				return "";
+			}).orElse("");
```

Comment:
In dem code ist es schwierig zu sehen was eigentlich gemacht wird. 

Der Status Text scheint wie bisher erstellt zu werden.

Anhand des Timestamp wird dann in den mandatorTrace nach dem passenden Eintrag gesucht, und von dem nur der Text übernommen.

Bitte entweder `stm[0]` einer Variable `timestampString` zuweisen, oder einen entsprechenden Kommentar einfügen. Damit das ein wenig einfacher zu verstehen ist.